src/t_string.c

```diff
@@ -99,7 +99,15 @@ void setGenericCommand(client *c, int flags, robj *key, robj *val, robj *expire,
         return;
     }
 
-    setkey_flags |= (flags & OBJ_KEEPTTL) ? SETKEY_KEEPTTL : 0;
+    /* Set the flags for setKey() function */
+
+    /* In the following 2 cases we set the 'SETKEY_KEEPTTL' flag to avoid removing the expiration entry 
+     * from 'db->expires' :
+     * 1. When 'OBJ_KEEPTTL' is set in the flags passed in.
+     * 2. When expire is not NULL, we avoid deleting so it can be updated later instead of first deleted and
+     * then created again.
+     */
```

Comment:
I meant most of this comment is redundant. IMO it is enough to describe only the part which is not obvious.

```suggestion
    /* When expire is not NULL, we avoid deleting so it can be updated later
     * instead of first deleted and then created again. */
```

Reply:
Ok. Remove these redundant comments